Go Node segfaults when binary is not found

## 🙂 Expected behavior

Raise a readable error

## 🫠 Actual behavior

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4e0660]

goroutine 34 [running]:
os.(*ProcessState).sys(...)
	/usr/local/go/src/os/exec_posix.go:91
os.(*ProcessState).Sys(...)
	/usr/local/go/src/os/exec.go:170
github.com/cartesi/rollups-node/internal/services.Service.Start({{0x51e257, 0xe}, 0x2afd, {0x52267e, 0x1e}, {0x0, 0x0, 0x0}, {0xc000142000, 0x4c, ...}}, ...)
	/home/gligneul/cartesi/rollups-node/internal/services/service.go:70 +0x220
github.com/cartesi/rollups-node/internal/services.Run.func1()
	/home/gligneul/cartesi/rollups-node/internal/services/service.go:124 +0xb9
created by github.com/cartesi/rollups-node/internal/services.Run in goroutine 1
	/home/gligneul/cartesi/rollups-node/internal/services/service.go:120 +0x305
exit status 2
```

## 🧪 Minimal test case

Start the Go node without one of the binaries in the PATH.

## 🌎 Environment

main

--- a/internal/services/service.go
+++ b/internal/services/service.go
@@ -7,6 +7,7 @@ package services
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net"
 	"os/exec"
@@ -64,16 +65,13 @@ func (s Service) Start(ctx context.Context) error {
 		}
 		return err
 	}
+
 	err := cmd.Run()
-	if err != nil {
-		exitCode := cmd.ProcessState.ExitCode()
-		signal := cmd.ProcessState.Sys().(syscall.WaitStatus).Signal()
-		if exitCode != 0 && signal != syscall.SIGTERM {
-			// only return error if the service exits for reason other than shutdown
-			return err
-		}
+
+	if ctx.Err() != nil {
+		return ctx.Err()
 	}
-	return nil
+	return err
 }
 
 // Ready blocks until the service is ready or the context is canceled.
@@ -121,7 +119,7 @@ func Run(ctx context.Context, services []Service) {
 			// cancel the context when one of the services finish
 			defer cancel()
 			defer wg.Done()
-			if err := service.Start(ctx); err != nil {
+			if err := service.Start(ctx); err != nil && !errors.Is(err, context.Canceled) {
 				msg := "main: service '%v' exited with error: %v\n"
 				logger.Error.Printf(msg, service.String(), err)
 			} else {

--- a/internal/services/service_test.go
+++ b/internal/services/service_test.go
@@ -60,7 +60,7 @@ func (s *ServiceTestSuite) TestServiceStops() {
 	// shutdown
 	cancel()
 	err := <-result
-	s.Nil(err, "service exited for the wrong reason: %v", err)
+	s.ErrorIs(err, context.Canceled, "service exited for the wrong reason: %v", err)
 }
 
 // Service should stop if timeout is reached and it isn't ready yet
@@ -85,7 +85,8 @@ func (s *ServiceTestSuite) TestServiceTimeout() {
 
 	// shutdown
 	cancel()
-	s.Nil(<-result, "service exited for the wrong reason: %v", err)
+	err = <-result
+	s.ErrorIs(err, context.Canceled, "service exited for the wrong reason: %v", err)
 }
 
 // Service should be ready soon after starting
@@ -110,7 +111,22 @@ func (s *ServiceTestSuite) TestServiceReady() {
 
 	// shutdown
 	cancel()
-	s.Nil(<-result, "service exited for the wrong reason: %v", err)
+	err = <-result
+	s.ErrorIs(err, context.Canceled, "service exited for the wrong reason: %v", err)
+}
+
+// Service should fail to start if its executable is not found in $PATH
+func (s *ServiceTestSuite) TestServiceFailsToStartIfNotInPath() {
+	service := Service{
+		Name:            "fake-service",
+		Path:            "wrong-path",
+		HealthcheckPort: s.servicePort,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := service.Start(ctx)
+	s.ErrorIs(err, exec.ErrNotFound, "service exited for the wrong reason: %v", err)
 }
 
 // Builds the fake-service binary and adds it to PATH

